packages/patternfly-4/react-core/src/demos/BulkSelectTable.md

```diff
@@ -0,0 +1,225 @@
+
+```js
+import React from 'react';
+import { Pagination, PaginationVariant, Title, Checkbox } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+
+class BulkSelectTableDemo extends React.Component {
+  constructor(props) {
+    this.state = {
+      res: [],
+      perPage: 20,
+      total: 100,
+      page: 1,
+      error: null,
+      loading: true,
+      selectedItems: [],
+      numSelected: 0,
+      isDropDownOpen: false,
+      isKebabOpen: false,
+      searchValue: ''
+    };
+
+    this.onSelect = (event, isSelected, rowId) => {
+      const { selectedItems } = this.state;
+      const rows = [...this.state.res];
+      const id = rows[rowId].id;
+      rows[rowId].selected = isSelected;
+      this.setState({
+        res: rows,
+        selectedItems: isSelected ? [...selectedItems, id] : selectedItems.filter(itemId => itemId !== id)
+      });
+    };
+
+    this.updateSelected = () => {
+      const { res, selectedItems } = this.state;
+      let rows = res.map(post => {
+        post.selected = selectedItems.includes(post.id);
+        return post;
+      });
+
+      this.setState({
+        res: rows
+      });
+    };
+
+    this.handleSelectClick = newState => {
+      if (newState === 'none') {
+        this.setState(
+          {
+            selectedItems: []
+          },
+          this.updateSelected
+        );
+      } else if (newState === 'page') {
+        let newRows = [];
+        let rows = this.state.res.map(post => {
+          const isSelected = post.selected;
+          newRows = isSelected ? [...newRows] : [...newRows, post.id];
+          post.selected = true;
+          return post;
+        });
+        this.setState(
+          {
+            selectedItems: this.state.selectedItems.concat(newRows)
+          },
+          this.updateSelected
+        );
+      } else {
+        const { total } = this.state;
+        let newRows = [];
+        for (var i = 1; i <= total; i++) newRows = [...newRows, i];
+
+        this.setState(
+          {
+            selectedItems: newRows
+          },
+          this.updateSelected
+        );
+      }
+    };
+
+    this.onDropDownToggle = isOpen => {
+      this.setState({
+        isDropDownOpen: isOpen
+      });
+    };
+
+    this.onDropDownSelect = event => {
+      this.setState({
+        isDropDownOpen: !this.state.isDropDownOpen
+      });
+    };
+  }
+
+  fetch(page, perPage) {
+    this.setState({ loading: true });
+    fetch(`https://jsonplaceholder.typicode.com/posts?_page=${page}&_limit=${perPage}`)
+      .then(resp => resp.json())
+      .then(resp => this.setState({ res: resp, perPage, page, loading: false }))
+      .then(() => this.updateSelected())
+      .catch(err => this.setState({ error: err, loading: false }));
+  }
+
+  componentDidMount() {
+    this.fetch(this.state.page, this.state.perPage);
+  }
+
+  renderPagination(variant = 'top') {
```

Comment:
Since we don't customize this `variant` property anywhere in the demo, it might help readability to simply use the string value 'top' directly when setting this property like `variant={'top'}`, then we could remove this param/default here.